### PR TITLE
Fix client side subcompilation errors

### PR DESCRIPTION
**Pull request by mfix-stripe**
Cc @ijjk

### CODE REVIEW

File: src/loader.js [hunk 1]
```diff
@@ -67,21 +67,11 @@ async function load(source) {
...
-    // https://github.com/pmmmwh/react-refresh-webpack-plugin/issues/176#issuecomment-683150213
-    global.$RefreshReg$ = previousRefreshReg || noop;
-    global.$RefreshSig$ = previousRefreshSig || (() => noop);
-    global.document = previousDocument || {
-      querySelector: noop,
-      querySelectorAll: noop,
-    };
-    global.Element = previousElement || class Element {};
+    let previousRequire = global.require;
+    global.require = previousRequire || require || __webpack_require__;
```
Comment (ijjk) [inline src/loader.js:74]:
```suggestion
    global.require = previousRequire || require || __non_webpack_require__;
```
Potentially wanna use the non-webpack variant here as it's an alias for `require` x-ref: https://webpack.js.org/api/module-variables/#__non_webpack_require__-webpack-specific

Comment (mfix-stripe) [reply src/loader.js]: @ijjk thanks for the tip! Will this cause errors in sub-compilation if, for example, the file that is `require`d is a TypeScript file?

Comment (mfix-stripe) [reply src/loader.js]: Rad, thanks so much!

**Comment by mfix-stripe** [comment]
PTAL @matv-stripe — you are the only one with context on this.

**Review by matv-stripe — APPROVED** [review]
i have some FUD about how complex this is getting, but not sure if there is a good way around that given the state of the ecosystem

**Comment by mfix-stripe** [comment]
> i have some FUD about how complex this is getting, but not sure if there is a good way around that given the state of the ecosystem

@matv-stripe same, but also I think this PR eliminates some hacks we already have in place. I figure this is a 2 steps forward one step back scenario, which is net positive. Do you agree?

**Pull request merged**